packages/fxa-settings/src/components/Ready/index.tsx

```diff
@@ -4,14 +4,44 @@
 
 import React from 'react';
 import { RouteComponentProps } from '@reach/router';
-import { Localized } from '@fluent/react';
+import { FtlMsg } from 'fxa-react/lib/utils';
 import { ReactComponent as PulseHearts } from './account-verified.svg';
 import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
 
 type ReadyProps = {
   serviceName?: string;
   continueHandler?: Function;
-  viewName: string;
+  viewName: viewNameType;
+};
+
+type viewNameType =
+  | 'signin-confirmed'
+  | 'signin-verified'
+  | 'reset-password-confirmed'
+  | 'reset-password-verified'
+  | 'reset-password-with-recovery-key-verified';
+
+const getTemplateValues = (viewName: viewNameType) => {
+  let templateValues: { headerText: string; headerId: string } = {
+    headerText: '',
+    headerId: '',
+  };
+  // I use the switch case here instead of just a hash because there's a lot of duplication,
+  // and I think the fallthrough structure is really intuitive! Open to feedback on this --
+  // if it seems unclear to people, that's a good reason to change it
```

Comment:
Nit: this feels more like a comment to leave on a self-review rather than in the code. 

The pattern seems good to me though given how we're using the `Ready` component so far, though I think an alternative could be to use `FtlMsgResolver` to grab the translated string and pass it into `Ready` so we don't have to have this switch at all, like this:

```
const localizedHeaderText = ftlMsgResolver.getMsg(
  'fxa-sign-in-complete-header',
  'Sign-in confirmed'
);
```

Then pass it in and receive it:

```
const Ready = ({
  serviceName = 'Account Settings',
  continueHandler,
  viewName,
  localizedHeaderText
}: ReadyProps & RouteComponentProps) => {
  ...
  <h1 className="card-header">
    {localizedHeaderText}
  </h1>
```

Either way I think we _really_ need Storybook states for this component. Your approach might make the Storybook states more easily documented, e.g. we pass in the `viewName` and let the component figure out text and l10n instead of us having wrappers where we pass in `localizedHeaderText`. I'd like to see what it can look like at every possibility for documentation purposes, because this component is likely to continue to grow in complexity. I'm fine with either approach 👍

Reply:
Thank you for the explanation of how to use the FtlMsgResolver! That's really helpful! I see what you mean about the relative ease-of-use in Storybook for each approach tho. I'll add those states into Storybook!